// draft-css/Gruntfile.js
'use strict';

module.exports = function(grunt) {

    // Project configuration.
    grunt.initConfig({
        pkg: grunt.file.readJSON('package.json'),
        banner: '/*!\n' +
                ' * Draft v<%= pkg.version %>\n' +
                ' * Copyright <%= grunt.template.today("yyyy") %> <%= pkg.author %>\n' +
                ' * Licensed under the terms of the <%= pkg.license %> license.\n' +
                ' */\n',
        usebanner: {
            dist: {
                options: {
                    position: 'top',
                    banner: '<%= banner %>'
                },
                files: {
                    src: ['dist/css/<%= pkg.name %>-<%= pkg.version %>.min.css']
                }
            }
        },
        sass: {
            dev: {
                options: {
                    outputStyle: 'expanded'
                },
                files: {
                    'public/css/<%= pkg.name %>.css': 'src/scss/main.scss'
                }
            },
            dist: {
                options: {
                    outputStyle: 'compressed'
                },
                files: {
                    'dist/css/<%= pkg.name %>-<%= pkg.version %>.min.css': 'src/scss/main.scss'
                }
            }
        },
        watch: {
            sass: {
                files: ['src/scss/**/*.scss'],
                tasks: ['sass:dev']
            },
            js: {
                files: ['src/js/**/*.js'],
                tasks: ['uglify:dev']
            }
        },
        copy: {
            dist: {
                expand: true,
                cwd: 'src/',
                src: '*.html',
                dest: 'dist/'
            }
        }
    });

    // Load tasks.
    grunt.loadNpmTasks('grunt-banner');
    grunt.loadNpmTasks('grunt-sass');
    grunt.loadNpmTasks('grunt-contrib-watch');
    grunt.loadNpmTasks('grunt-contrib-copy');

    // Default task.
    grunt.registerTask('default', ['watch']);
    grunt.registerTask('dist', ['sass:dist', 'usebanner:dist', 'copy:dist']);

};
